lib/DocumentManagerInterface.php

```diff
@@ -122,4 +124,56 @@ public function createQuery($query, $locale = null, array $options = []);
      * http://doctrine-phpcr-odm.readthedocs.org/en/latest/reference/query-builder.html
      */
     public function createQueryBuilder();
+
+    /**
+     * Retrieve the document inspector.
+     *
+     * NOTE: In the future this will be a factory for retrieving inspectors
+     * for a given object/class.
```

Comment:
In the ORM the equivilent of the document instpector is the
EntityManager/DocumentManager, which is used to introspect the document.

The reason I introduced this class was to avoid API bloat for the
DocumentManager and to eventually allow a flexible way to introduce
introspection methods.

The problem with the Douments "knowing" about this stuff is that the
end-user needs to implement that stuff. If we want to know which
webspace a document belongs to we need to make `->getWebspaceName()` a
mandatory method on all documents, etc.

On Thu, Apr 14, 2016 at 01:00:31AM -0700, Daniel Rotter wrote:
>    In [1]lib/DocumentManagerInterface.php:
> 
>  > @@ -122,4 +124,56 @@ public function createQuery($query, $locale = null, array $options = []);
>  >       * http://doctrine-phpcr-odm.readthedocs.org/en/latest/reference/query-builder.html
>  >       */
>  >      public function createQueryBuilder();
>  > +
>  > +    /**
>  > +     * Retrieve the document inspector.
>  > +     *
>  > +     * NOTE: In the future this will be a factory for retrieving inspectors
>  > +     * for a given object/class.
> 
>    Ok, that's why there is this one factory class at the moment... Don't like
>    the idea, since the DocumentInspector is just a temporary hack IMO. The
>    document should be able to tell stuff like that on its own, there is also
>    no real equivalent to the DocumentInspector in the ORM.
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu/sulu-document-manager/pull/66#discussion_r59676182
>    2. https://github.com/sulu/sulu-document-manager/pull/66/files/6e7fab9301c97bff800cfff05d913f78e432ded1#r59676182
>    3. https://github.com/sulu/sulu-document-manager/pull/66/files/6e7fab9301c97bff800cfff05d913f78e432ded1#r59676182